extensions/filesystem/libuv/src/unix/fs.c

```diff
@@ -620,7 +620,39 @@ static int uv__fs_readdir(uv_fs_t* req) {
     if (dirent->name == NULL)
       goto error;
 
-    dirent->type = uv__fs_get_dirent_type(res);
+#if TO_HAIKU
```

Comment:
This file is part of libuv... it seems they think they support Haiku (notice the `defined(__HAIKU__)` cases in this file).

I'm hoping to not need to have a patched version of the files from libuv.  Perhaps they would accept this modification as a PR (or have feedback on it)?  Just changing the TO_HAIKU to defined(__HAIKU__) would probably be enough to start the discussion.

Perhaps also posting a note on this issue would send notification to parties who've previously expressed an interest in Haiku/libuv:

  https://github.com/libuv/libuv/issues/1414